internal/commands/gateway/list.go

```diff
@@ -0,0 +1,72 @@
+package gateway
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/commands"
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/output"
+	"github.com/UpCloudLtd/upcloud-cli/v3/internal/ui"
+	"github.com/UpCloudLtd/upcloud-go-api/v6/upcloud"
+	"github.com/jedib0t/go-pretty/v6/text"
+)
+
+// ListCommand creates the "gateway list" command
+func ListCommand() commands.Command {
+	return &listCommand{
+		BaseCommand: commands.New("list", "List gateways", "upctl gateway list"),
+	}
+}
+
+type listCommand struct {
+	*commands.BaseCommand
+}
+
+// ExecuteWithoutArguments implements commands.NoArgumentCommand
+func (c *listCommand) ExecuteWithoutArguments(exec commands.Executor) (output.Output, error) {
+	svc := exec.All()
+	gateways, err := svc.GetGateways(exec.Context())
+	if err != nil {
+		return nil, err
+	}
+
+	rows := []output.TableRow{}
+	for _, gtw := range gateways {
+		rows = append(rows, output.TableRow{
+			gtw.UUID,
+			gtw.Name,
+			gtw.Routers,
+			gtw.OperationalState,
+			gtw.Zone,
+		})
+	}
+
+	// For JSON and YAML output, passthrough API response
+	return output.MarshaledWithHumanOutput{
+		Value: gateways,
+		Output: output.Table{
+			Columns: []output.TableColumn{
+				{Key: "uuid", Header: "UUID", Colour: ui.DefaultUUUIDColours},
+				{Key: "name", Header: "Name"},
+				{Key: "routers", Header: "Routers", Format: formatRouters},
```

Comment:
This could include features list as well, but we can add that later when we have more than `NAT` as possible value there.